### PR TITLE
audacity: fix build on < 10.7 with libcxx

### PR DESCRIPTION
**Pull request by kencu**
the software author(s) have done quite a bit of work to enable
a built on SnowLeopard which I greatly appreciate, however when
MacPorts is set up to use libc++ and when building with a current
clang system, these specific fixes now cause the build to fail.

This patch disables those fixes if MacPorts is setup to use libc++.

As the fixes were only enabled for SnowLeopard, disabling them universally
if MacPorts is setup for libc++ is desirable to allow detection of meaningful changes
in the source file, and will have no effect on any systems other than < 10.7

```
$ port -v installed audacity
The following ports are currently installed:
  audacity @2.2.2_1 (active) platform='darwin 10' archs='x86_64' date='2019-01-10T20:51:01-0800'
```

### CODE REVIEW

**Comment by mojca** [comment]
@RJBV (no idea why the bot would not assign the maintainer)

File: audio/audacity/files/patch-src-memoryx-disable-snowleopard-fix.diff [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/src/MemoryX.h.old	2017-11-18 15:43:36.000000000 -0800
++++ b/src/MemoryX.h	2017-11-18 15:43:51.000000000 -0800
+@@ -12,7 +12,7 @@
+ // std:: containers knowing about rvalue references
+ #undef __AUDACITY_OLD_STD__
+ 
+-#if defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED <= __MAC_10_6
```
Comment (RJVB) [inline audio/audacity/files/patch-src-memoryx-disable-snowleopard-fix.diff]: Why is this required? Are you expecting that someone might try to build against libstdc++ on 10.7+ (other than you during testing)?

(This does remind me I should pick up my work on a `-stdlib=libc++` option for GCC again!)

**Comment by RJVB** [comment]
Sorry, I wasn't sufficiently awake it seems. The patch is added by this PR, not modified. So that's fine.
